src/plone/app/robotframework/i18n.py

```diff
@@ -1,25 +1,34 @@
 # -*- coding: utf-8 -*-
-import os
-
-from Products.CMFCore.utils import getToolByName
 from plone.app.robotframework.remote import RemoteLibrary
 from plone.app.robotframework.utils import disableCSRFProtection
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
 from zope.component.hooks import getSite
 from zope.globalrequest import getRequest
 from zope.i18n import translate
 
+import os
+
 
 class I18N(RemoteLibrary):
 
     def set_default_language(self, language=None):
         """Change portal default language"""
         disableCSRFProtection()
-        portal = getSite()
-        portal_languages = getToolByName(portal, 'portal_languages')
         if language is None:
             language = os.environ.get('LANGUAGE') or 'en'
-        setattr(portal, 'language', language)
-        portal_languages.setDefaultLanguage(language)
+        try:
+            from Products.CMFPlone.interfaces import ILanguageSchema
+        except ImportError:
```

Comment:
@jensens this code is for Plone 4 and should be removed.

Reply:
Yes, I just made it work again, but, as stated above, I may have missed some case, or I may be too careful. Please file a new issue or provide a PR!

Reply:
it depends on what you want; I think we can move on on these core packages leaving Plone 4 behind. let me know and I will add a PR with the clean up.

Reply:
Yes, great! Plone 4.3 coredev already sits on the 1.2.x branch, so master targets already Plone 5.x only.